src/sagas/wallet.js

```diff
@@ -447,6 +448,43 @@ export function* listenForWalletReady(wallet) {
   }
 }
 
+/**
+ * An abstraction effect to call wallet's checkAddressMine method.
+ * This effect applies a retry strategy and throws a fatal error
+ * after max retry attempts.
+ *
+ * @param {Object} wallet Either fullnode or wallet-service wallet.
+ * @param {string[]} txAddresses A collection of addresses to check.
+ * @returns {{ [address: string]: boolean }} A map of type address:isMine.
+ */
+function* checkAddressMine(wallet, txAddresses) {
+  const maxRetries = 5;
+  const resultHandler = async (cb) => {
+    try {
+      const success = await cb();
+      return { success };
+    } catch (error) {
+      return { error }
+    }
+  };
+
+  let result = null;
+  for (let i = 0; i < maxRetries; i++) {
+    result = yield call(resultHandler, async () => await wallet.checkAddressesMine.bind(wallet)([...txAddresses]));
```

Comment:
Why do you need this resultHandler? Can't you just try/catch here?

Reply:
Done at https://github.com/HathorNetwork/hathor-wallet-mobile/pull/444/commits/89c53c7f1c73034086e9cc5e4f59375df32fcf26

Reply:
I would agree that it improves code readability if this wrapper was also being used somewhere else and it was a pattern in our error handling, but imho it only adds complexity to an already clear construct which is the try/catch

My suggestion would be to just add the try catch here until it reaches maxRetries, throw if it reaches it and let the caller handle the error as this is what all other methods in the code do